src/SlashCommands.js

```diff
@@ -282,7 +282,74 @@ var commands = {
             }
         }
         return reject(this.getUsage());
-    })
+    }),
+
+    // Verify a user, device, and pubkey tuple
+    verify: new Command("verify", "<userId> <deviceId> <deviceSigningKey>", function(room_id, args) {
+        if (args) {
+            var matches = args.match(/^(\S+) +(\S+) +(\S+)$/);
+            if (matches) {
+                const userId = matches[1];
+                const deviceId = matches[2];
+                const fingerprint = matches[3];
+
+                const device = MatrixClientPeg.get().getStoredDevice(userId, deviceId);
+                if (!device) {
+                    return reject(`Unknown (user, device) pair: (${userId}, ${deviceId})`);
+                }
+
+                if (device.isVerified()) {
+                    if (device.getFingerprint() === fingerprint) {
+                        return reject(`Device already verified!`);
+                    } else {
+                        return reject(`WARNING: Device already verified, but keys do NOT MATCH!`);
+                    }
+                }
+
+                if (device.getFingerprint() === fingerprint) {
+                    const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+                    Modal.createDialog(QuestionDialog, {
+                        title: "Approve verified device",
+                        description: (
+                            <div>
+                                <p>
+                                    The signing key in your slash command matches the signing key you
+                                    received for this user and device!
+                                </p>
+                                <div className="mx_UserSettings_cryptoSection">
+                                    <ul>
+                                        <li><label>User ID:</label> <span>{ userId }</span></li>
+                                        <li><label>Device name:</label> <span>{ device.getDisplayName() }</span></li>
+                                        <li><label>Device ID:</label>   <span><code>{ device.deviceId}</code></span></li>
+                                        <li><label>Device fingerprint:</label>  <span><code><b>{ device.getFingerprint() }</b></code></span></li>
+                                        <li><label>Device key:</label>  <span><code><b>{ device.getIdentityKey() }</b></code></span></li>
+                                    </ul>
+                                </div>
+                                <p>
+                                    If you would like to accept this device as verified, press accept!
```

Comment:
Given the text on the button is "Accept this verified key", I think this text is redundant.